src/Storages/MergeTree/MergeTreeDataPartWriterOnDisk.cpp

```diff
@@ -378,11 +387,16 @@ void MergeTreeDataPartWriterOnDisk::calculateAndSerializeSkipIndices(const Block
                     writeBinaryLittleEndian(1UL, marks_out);
             }
 
+            ProfileEventTimeIncrement<Microseconds> watch(ProfileEvents::MergeTreeDataWriterSecondaryIndicesCalculationMicroseconds);
+
             size_t pos = granule.start_row;
             skip_indices_aggregators[i]->update(skip_indexes_block, &pos, granule.rows_to_write);
             if (granule.is_complete)
                 ++skip_index_accumulated_marks[i];
+
+            index_build_us += watch.elapsed<Microseconds>();
         }
+        LOG_DEBUG(log, "Spent {} ms calculating index {} for the part {}", index_build_us / 1000, skip_indices[i]->index.name, data_part->name);
```

Comment:
I think, it's better to print this message at the end of writing, e.g. in `finishSkipIndicesSerialization` because this method may be called several times, e.g. during merges and mutations.